pkg/kubernetes/kubernetes.go

```diff
@@ -40,57 +38,46 @@ func New(s v1alpha1.Spec) *Kubernetes {
 	return &k
 }
 
-// Manifest describes a single Kubernetes manifest
-type Manifest map[string]interface{}
-
-func (m Manifest) Kind() string {
-	return m["kind"].(string)
-}
-
-func (m Manifest) Name() string {
-	return m["metadata"].(map[string]interface{})["name"].(string)
-}
-
-func (m Manifest) Namespace() string {
-	return m["metadata"].(map[string]interface{})["namespace"].(string)
-}
-
-// Reconcile receives the raw evaluated jsonnet as a marshaled json dict and
-// shall return it reconciled as a state object of the target system
-func (k *Kubernetes) Reconcile(raw map[string]interface{}, objectspecs []*regexp.Regexp) (state []Manifest, err error) {
-	docs, err := walkJSON(raw, "")
-	out := make([]Manifest, 0, len(docs))
-	if err != nil {
-		return nil, errors.Wrap(err, "flattening manifests")
-	}
-	for _, d := range docs {
-		m := objx.New(d)
-		if k != nil && !m.Has("metadata.namespace") {
-			m.Set("metadata.namespace", k.Spec.Namespace)
-		}
-		out = append(out, Manifest(m))
-	}
-
-	if len(objectspecs) > 0 {
-		out = funk.Filter(out, func(i interface{}) bool {
-			p := objectspec(i.(Manifest))
-			for _, o := range objectspecs {
-				if o.MatchString(strings.ToLower(p)) {
+// Build takes the raw evaluated jsonnet as a marshaled json dict, along with some
+// <kind>/<name> matching regular expressions. It will return a list of Manifests, which
+// it finds by recursively walking the jsonnet structure, and filtering each manifest
+// against the matchers.
+//
+// In addition, we sort the manifests to ensure the order is consistent in each
+// show/diff/apply cycle. This isn't necessary, but it does help users by producing
+// consistent diffs.
+func (k *Kubernetes) Build(raw map[string]interface{}, kindNameMatchers []*regexp.Regexp) []Manifest {
```

Comment:
Btw, please avoid renaming functions, I think `Reconcile` worked well and was a more descriptive name than `Build` (what are we building here?) :D

Reply:
We would be building manifests, from the jsonnet source.

I'm totally fine not renaming, this name just confused me and a teammate. Reconcile means to make two things consistent, by removing differences. This function directly translates jsonnet into kubernetes objects, so there shouldn't be any differences to reconcile?

Primarily the rename helped to avoid collision with the concept of kubernetes operators. We were really puzzled when we saw reconcile, we thought we'd found a controller!

Reply:
Oh, I wasn't aware of this! I am not a native english speaker, apparently the dictionary let me down :D

And if the name is already reserved for controllers, we really shouldn't use it.

Still, `Build` seems not descriptive enough to me, what about `FindObjects` because that's actually what we do, we search for Kubernetes objects.

PS: Is there a better word for Kubernetes object? Manifest? API Resource? Something else?

Reply:
That's cool! Naming is hard!

Build is very generic, and could be better. Probably FindManifests is an appropriate name? I think Manifest is the type this function returns, and that term is used when talking about the yamls that are being submitted to the kubernetes API, so seems appropriate?

Reply:
Sounds cool! Like that